pattern/constraint/thing/ValueConstraint.java

```diff
@@ -195,6 +198,16 @@ public FunctionalIterator<AlphaEquivalence> alphaEquals(ValueConstraint<?> that)
                 .flatMap(a -> a.alphaEqualIf(this.value.equals(that.value)));
     }
 
+    public boolean inconsistentWith(ValueConstraint<?> valueConstraint) {
+        if (valueConstraint.predicate == EQ) {
+            return !isConsistentWithEquality(valueConstraint);
+        } else { // TODO: implement inequality compatibility when useful
+            return true;
+        }
+    }
+
+    protected abstract boolean isConsistentWithEquality(ValueConstraint<?> conclusionValueConstraint);
```

Comment:
this should be without a modifier, ie. `package-private`, along with all the implementations below